src/platform/nrf_pca10028/src/main.rs

```diff
@@ -0,0 +1,267 @@
+//! pins, LEDs, and buttons. Currently the kernel provides
+//! application timers, and GPIO. It will provide a console
+//! once the UART is fully implemented and debugged. The
+//! application GPIO pins are:
+//!
+//!   0 -> LED1 (pin 21)
+//!   1 -> LED2 (pin 22)
+//!   2 -> LED3 (pin 23)
+//!   3 -> LED4 (pin 24)
+//!   5 -> BUTTON1 (pin 17)
+//!   6 -> BUTTON2 (pin 18)
+//!   7 -> BUTTON3 (pin 19)
+//!   8 -> BUTTON4 (pin 20)
+//!   9 -> P0.01   (bottom left header)
+//!  10 -> P0.02   (bottom left header)
+//!  11 -> P0.03   (bottom left header)
+//!  12 -> P0.04   (bottom left header)
+//!  12 -> P0.05   (bottom left header)
+//!  13 -> P0.06   (bottom left header)
+//!  14 -> P0.19   (mid right header)
+//!  15 -> P0.18   (mid right header)
+//!  16 -> P0.17   (mid right header)
+//!  17 -> P0.16   (mid right header)
+//!  18 -> P0.15   (mid right header)
+//!  19 -> P0.14   (mid right header)
+//!  20 -> P0.13   (mid right header)
+//!  21 -> P0.12   (mid right header)
+//!
+//!  Author: Philip Levis <pal@cs.stanford.edu>
+//!  Author: Anderson Lizardo <anderson.lizardo@gmail.com>
+//!  Date: August 18, 2016
+
+#![crate_name = "nrf_pca10028"]
+#![no_std]
+#![no_main]
+#![feature(core_intrinsics,lang_items)]
+
+extern crate cortexm0;
+extern crate drivers;
+extern crate hil;
+extern crate nrf51;
+extern crate main;
+extern crate support;
+
+use drivers::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use drivers::timer::TimerDriver;
+use nrf51::timer::TimerAlarm;
+use nrf51::timer::ALARM1;
+use hil::gpio::GPIOPin;
+
+// The nRF51 DK LEDs (see back of board)
+const LED1_PIN:  usize = 21;
+const LED2_PIN:  usize = 22;
+const LED3_PIN:  usize = 23;
+const LED4_PIN:  usize = 24;
+
+// The nRF51 DK buttons (see back of board)
+const BUTTON1_PIN: usize = 17;
+const BUTTON2_PIN: usize = 18;
+const BUTTON3_PIN: usize = 19;
+const BUTTON4_PIN: usize = 20;
+
+pub mod systick;
+
+unsafe fn load_process() -> &'static mut [Option<main::process::Process<'static>>] {
+    use core::intrinsics::{volatile_load,volatile_store};
+    extern {
+        /// Beginning of the ROM region containing app images.
+        static _sapps : u8;
+    }
+
+
+    #[link_section = ".app_memory"]
+    static mut MEMORY: [u8; 8192] = [0; 8192];
+    static mut PROCS: [Option<main::process::Process<'static>>; 1] = [None];
+
+    let addr = &_sapps as *const u8;
+
+    // The first member of the LoadInfo header contains the total size of
+    // each process image. A sentinel value of 0 (invalid because it is 
+    // smaller than the header itself) is used to mark the end of the list 
+    // of processes.
+    let total_size = volatile_load(addr as *const usize);
+    if total_size != 0 {
+        volatile_store(&mut PROCS[0], Some(main::process::Process::create(addr,
+                                                                          total_size, &    mut MEMORY)));
+    }
+    &mut PROCS
+}
+
+pub struct Platform {
+    //chip: nrf51::chip::Nrf51,
```

Comment:
Why? The SAM4L has this code. Is this because the NRF can currently only support a single process? What if that changes? It seemed like keeping the code as parallel as possible was desirable.